### ISSUE TITLE
Bug: IncomingRequest.php getVar() 

### ISSUE THREAD

**Issue by lenuse** [codeigniter4/CodeIgniter4#4381]
**Direction**
The getVar method cannot get $_REQUEST under a GET request 

**Describe the bug**
When the http content-type is json, send a get request, the request parameters cannot be obtained, and the program issues an exception 
![C24864D6-8178-4fe4-AA76-933DCB50A2A9](https://user-images.githubusercontent.com/12166648/109593164-71fb7280-7b4b-11eb-9a68-d7906a05aded.png)

**CodeIgniter 4 version**
4.1.1

**Affected module(s)**
getVar method

**Expected behavior, and steps to reproduce if appropriate**
When $this->body is null, you should get the value in $_REQUEST 

**Context**
 - OS:  Windows 10
 - Web server  nginx all
 - PHP version 7.4.3

**Comment by luispastendev** [comment]
When you receive json content you can get it with `$this->request->getJson(true)` which will get the content and convert it into an associative array with that you could access the values of your request 

```php
$body = $this->request->getJSON(true);
print_r($body['param']);
```

**Comment by lenuse** [comment]
```js
$.ajax({
   url: "/api/products?type=1&page=1&limit=20",
    headers: {
        Accept: "application/json; charset=utf-8"
    },
    type: "get",
    success: function (data) {
    }
});
```
```php
class products extends \CodeIgniter\Controller
{
    // url is /api/products
    public function products() {
        $this->request->getVar('limit');
    }
}

```
I want to use `$this->request->getVar('limit')` but but the program throws an exception。

**Comment by luispastendev** [comment]
@lenuse what I see what you want to get is actually a query string not a json, `getVar` is not going to serve you for that, what you should do is do something like the following:

```php
parse_str($this->request->uri->getQuery(),$output);
echo $output['limit'];
```

see: 
- https://codeigniter.com/user_guide/libraries/uri.html#query
- https://www.php.net/manual/es/function.parse-str.php

**Comment by lenuse** [comment]
In version 4.0.4, using the getVar() method to obtain query string parameters is allowed. But after adding the content-type judgment, the method is incompatible. And 4.0.5 is a minor version update, this should not happen. I have now updated the code of my production environment.

**Comment by lenuse** [comment]
If it is not compatible, please close this bug

**Comment by luispastendev** [comment]
you can close the issue or some admin to do it @paulbalandan

**Comment by luispastendev** [comment]
@lenuse I gave myself the task of doing a quick test but apparently if getVar should work to receive the parameters of the query string  

```php
// .../test?foo=baz&bar=foo&limit=11

public function test(){
    $val = $this->request->getVar('limit');
    echo json_encode(['limit' => $val]);
}
```

![imagen](https://user-images.githubusercontent.com/4117596/110972199-58afae80-8321-11eb-81cd-fd1f8a974e98.png)

I'm testing in the develop branch, I don't know if I change something in this part of the core, at least it works fine for me.

**Comment by paulbalandan** [comment]
@lenuse if you can confirm that the previous behavior in 4.0.4 is working in the latest develop as @luispastendev tested, then this no longer a bug

**Comment by lenuse** [comment]
I suggest compatibility with the previous methods, for example: 
```php
public function getVar($index = null, $filter = null, $flags = null)
	{
		if (strpos($this->getHeaderLine('Content-Type'), 'application/json') !== false && !is_null($this->body))
		{
			if (is_null($index))
			{
				return $this->getJSON();
			}

			if (is_array($index))
			{
				$output = [];
				foreach ($index as $key)
				{
					$output[$key] = $this->getJsonVar($key, false, $filter, $flags);
				}
				return $output;
			}

			return $this->getJsonVar($index, false, $filter, $flags);
		}

		return $this->fetchGlobal('request', $index, $filter, $flags);
	}
```

**Comment by MGatner** [comment]
All I see changed in your suggestion is adding the check for `null` body. I don't see any changes since 4.0.4 that would have caused this behavior to change. I'm inclined to close this as a non-issue, but if you would like to submit a PR with the proposed change and a test case demonstrating the issue I would be glad to receive it.

**Comment by paulbalandan** [comment]
Fixed by #4493
